classes/class-wc-connect-custom-surcharge.php

```diff
@@ -39,50 +57,48 @@ public static function add_us_co_rdf( $cart = null ) {
 				return;
 			}
 
-			if ( ! is_array( $cart->get_cart_contents_taxes() ) || 0 === count( $cart->get_cart_contents_taxes() ) ) {
+			// Do not apply RDF if the customer is not in US Colorado.
+			if (
+				'US' !== WC()->customer->get_shipping_country()
+				|| 'CO' !== WC()->customer->get_shipping_state()
+			) {
 				return;
 			}
 
-			// Do not apply the fee if all products are virtual.
-			if ( ! $cart->needs_shipping() ) {
+			// Do not apply RDF when every item in order is exempt from Colorado sales tax.
+			if (
+				! is_array( $cart->get_cart_contents_taxes() )
+				|| 0 === count( $cart->get_cart_contents_taxes() )
+			) {
 				return;
 			}
-	
-			// Do not apply the fee if all shipping methods use Local Pickup.
-			if ( 0 === count( array_diff( wc_get_chosen_shipping_method_ids(), apply_filters( 'woocommerce_local_pickup_methods', array( 'legacy_local_pickup', 'local_pickup' ) ) ) ) ) {
+
+			// Do not apply RDF if all shipping methods use Local Pickup.
+			if ( 0 === count(
+				array_diff(
+					wc_get_chosen_shipping_method_ids(),
+					/**
+					 * Filters local pickup shipping methods.
+					 * Copied from WooCommerce core to maintain compatability.
+					 *
+					 * @since 6.8.0
+					 * @param string[] $local_pickup_methods Local pickup shipping method IDs.
+					 */
+					apply_filters( 'woocommerce_local_pickup_methods', array( 'legacy_local_pickup', 'local_pickup' ) )
+				)
+			) ) {
 				return;
 			}
 
-			if ( 'US' !== WC()->customer->get_shipping_country() || 'CO' !== WC()->customer->get_shipping_state() ) {
+			// Do not apply RDF if all products are virtual.
+			if ( ! $cart->needs_shipping() ) {
 				return;
 			}
 
-			/**
-			 * Filter for manipulate the custom surcharge.
-			 * Set the 'enabled' key to `false` to disable the custom surcharge.
-			 *
-			 * @since 2.8.6
-			 *
-			 * @param array   Custom surcharge info.
-			 * @param WC_Cart WooCommerce cart object.
-			 */
-			$fee_info = apply_filters(
-				'wc_services_apply_us_co_retail_delivery_fee',
-				array(
-					'enabled' => false,
-					'value'   => 0.29,
-					'text'    => __( 'Retail Delivery Fee', 'woocommerce_services' ),
-				),
-				$cart
-			);
-			
-			if (
-				isset( $fee_info['enabled'] ) && true === $fee_info['enabled'] &&
-				isset( $fee_info['text'] ) &&
-				isset( $fee_info['value'] ) && is_numeric( $fee_info['value'] )
-			) {
-				$cart->add_fee( $fee_info['text'], floatval( $fee_info['value'] ), true, 'standard' );
-			}
+			// As of July 1, 2024 till June 30, 2025 RDF is 29 cents per order
+			// RDF is subject to sales tax.
+			// https://www.avalara.com/blog/en/north-america/2022/10/what-you-need-to-know-about-the-colorado-retail-delivery-fee-now.html.
+			$cart->add_fee( __( 'Retail Delivery Fee', 'woocommerce_services' ), 0.29, true, 'standard' );
```

Comment:
@bartech , the reason I was putting the filter here is to let the user change the fee value easily through filter if the new fee is coming. Do you see any issue in the future if we put the filter here?

Reply:
We can update the fee. User can also based on this method logic create their own and just make our disabled.

Reply:
Yeah I guess they can update the method. Although, I'm still unsure if all user have the knowledge as we do.

Reply:
You can wrap Fee Name and Fee Value in filter as it was before as array. I was thinking about it too and was on the fence if this will be used by the merchant and if you think it will let's do that. Change I did was mostly to move bool flag filter to the top of the method instead of doing whole work of checking the logic to ditch everything at the end when enabled/disabled was a part of data for `add_fee`

Reply:
ok, let me add that.